src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -9957,12 +9972,12 @@ static void Main()
                 """;
             var comp = CreateCompilation(source, targetFramework: TargetFramework.Net80);
             comp.VerifyEmitDiagnostics(
-                // 0.cs(6,24): error CS0416: 'T': an attribute argument cannot use type parameters
+                // (6,24): error CS0416: 'T': an attribute argument cannot use type parameters
                 //     [CollectionBuilder(typeof(T), "ToString")]
                 Diagnostic(ErrorCode.ERR_AttrArgWithTypeVars, "typeof(T)").WithArguments("T").WithLocation(6, 24),
-                // 0.cs(19,45): error CS1061: 'Container<string>.MyCollection' does not contain a definition for 'Add' and no accessible extension method 'Add' accepting a first argument of type 'Container<string>.MyCollection' could be found (are you missing a using directive or an assembly reference?)
+                // (19,44): error CS9174: Cannot initialize type 'Container<string>.MyCollection' with a collection expression because the type is not constructible.
```

Comment:
:memo: I don't love this error message, but that's the existing catch all message we have...

Reply:
If we add back the *collection initializer* case to `GenerateImplicitConversionErrorForCollectionExpression`, would we get the expected binding error for `Add()`?